extras/indexing/src/main/java/mvm/rya/indexing/accumulo/VisibilityBindingSet.java

```diff
@@ -0,0 +1,52 @@
+package mvm.rya.indexing.accumulo;
+
+import static com.google.common.base.Preconditions.checkNotNull;
+
+import java.util.ArrayList;
+import java.util.Collection;
+
+import org.openrdf.query.BindingSet;
+
+import mvm.rya.indexing.external.BindingSetDecorator;
+
+/**
+ * Decorates a {@link BindingSet} with a collection of visibilities.
+ */
+public class VisibilityBindingSet extends BindingSetDecorator {
+	private static final long serialVersionUID = 1L;
+	private Collection<String> visibilitySet;
+	
+	public VisibilityBindingSet(BindingSet set) {
+		this(set, new ArrayList<String>());
+	}
+	
+	public VisibilityBindingSet(BindingSet set, Collection<String> visibility) {
+		super(set);
+		this.visibilitySet = visibility;
+	}
+
+	public Collection<String> getVisibility() {
+		return visibilitySet;
+	}
+	
+	public void setVisiblity(Collection<String> visibilitySet) {
+		this.visibilitySet = checkNotNull(visibilitySet);
+	}
+	
+	@Override
+	public boolean equals(Object o) {
```

Comment:
This will throw a ClassCastException if o is not of type VisibilityBindingSet.

public boolean equals(Object o) {
    if(this == o) {
        return true;
    } else if(o instanceof VisibilityBindingSet) {
        // CHECK HERE
    }
    return false;
}